File: ProvidedClients/JavascriptClient/Snake.js
class Snake {
    head = [];
    segments = new Array();
    length = 1;
    name = "";
    kidcount = 0;

    constructor(address, name) {
        console.log("creating new snake: " + name);
        console.log(address);
        this.head = address;
        this.segments.push(address);
        this.name = name;
        console.log("snake is finished: " + name);
    }
}

exports.Snake = Snake;

File: ProvidedClients/JuliaClient/Snake.jl
mutable struct Snake
    Head::Vector{Int};
    Segments::Vector{Vector{Int}}
    Length::Int;
    Name::String;
    KidCount::Int;

    function Snake(address::Vector{Int}, name::String)
        println("creating new snake: " * name);
        Segments = Vector{Vector{Int}}(undef, 0);
        push!(Segments, address);
        return new(address, Segments, Int.(1), name, 0);
        println("snake is finished: " * name);
    end
end
